tools/snp.sh

```diff
@@ -1081,42 +1062,134 @@ attest_guest() {
+  ssh_guest_command "./$Home/snpguest/target/release/snpguest fetch vcek pem ${cpu_code_name} . attestation-report.bin"
 
   # Verifies that ARK, ASK and VCEK are all properly signed
-  ssh_guest_command "./snpguest verify certs ."
+  ssh_guest_command "./$Home/snpguest/target/release/snpguest verify certs ."
 
   # Verifies the attestation-report trusted compute base matches vcek
-  ssh_guest_command "./snpguest verify attestation . attestation-report.bin"
+  ssh_guest_command "./$Home/snpguest/target/release/snpguest verify attestation . attestation-report.bin"
+  
+ if [ "$SVSM" = true ]; then 
+  
+	  # Use sev-snp-measure utility to calculate the expected measurement
+	  local expected_measurement=$(generate_snp_expected_measurement)
+	  echo -e "\nExpected Measurement (sev-snp-measure):  ${expected_measurement}"
+
+	  # Parse the measurement out of the snp report
+	  local snpguest_report_measurement=$(ssh_guest_command \
+		"./snpguest display report attestation-report.bin \
+		| tr '\n' ' ' \
+		| sed \"s|.*Measurement:\(.*\)Host Data.*|\1\n|g\" \
+		| sed \"s| ||g\"")
+
+	  # Remove any special characters and print the value
+	  snpguest_report_measurement=$(echo ${snpguest_report_measurement} | sed $'s/[^[:print:]\t]//g')
+	  echo -e "Measurement from SNP Attestation Report: ${snpguest_report_measurement}\n"
+
+	  # Compare the expected measurement to the guest report measurement
+	  [[ "${expected_measurement}" == "${snpguest_report_measurement}" ]] \
+		&& echo -e "The expected measurement matches the snp guest report measurement!" \
+		|| { >&2 echo -e "FAIL: measurements do not match"; return 1; }
+else 
+ 	 Verify_measurement
+fi
+}
+setup_svsm_guest_attestation() {
+
+   ssh_guest_command "
+   
+    # Update and install necessary packages
+    echo 'Updating package list...'
+    sudo apt-get update
+
+    echo 'Installing necessary packages...'
+    sudo apt-get install -y git build-essential libtss2-dev tpm2-tools
+
+    echo 'Installing Rust...'
+    source ""${HOME}"/.cargo/env" 2>/dev/null || true
+
+    curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs -sSf | sh -s -- -y
+    source ""${HOME}"/.cargo/env" 2>/dev/null
+    "    
+
+  # Check if 'snpguest' directory exists and remove it if it does
+  ssh_guest_command "
+    if [ -d \"snpguest\" ]; then
+        echo 'Directory snpguest exists. Removing it...'
+        rm -rf snpguest
+    else
+        echo 'Directory snpguest does not exist.'
+    fi
+"
+
+  # Clone the repository and build the project
+  ssh_guest_command "git clone https://github.com/virtee/snpguest.git"
+  ssh_guest_command "cd snpguest  &&  source \$HOME/.cargo/env &&  cargo build --release"
+}
 
-  # Use sev-snp-measure utility to calculate the expected measurement
-  local expected_measurement=$(generate_snp_expected_measurement)
-  echo -e "\nExpected Measurement (sev-snp-measure):  ${expected_measurement}"
+Verify_measurement(){
```

Comment:
Why do we have two separate verification functions? Wouldn't it make sense to just update the existing one?

Reply:
For this one he is using the igvmmeasure rather than the sev-snp-measure tool. You will need a separate function to calculate the svsm expected measurement, but then maybe put that function back into the original verify measurement.

Reply:
Exactly what @DGonzalezVillal is saying. That, or put a flag in the function for verify, specifying you want to use SVSM logic. We already have the flag. It wouldn't be hard to do a check and adjust the logic.